common/rpc/encryption/localStoreTlsFactory.go

```diff
@@ -241,3 +317,69 @@ func newClientTLSConfig(clientProvider ClientCertProvider, isAuthRequired bool,
 		!clientProvider.DisableHostVerification(isWorker),
 	), nil
 }
+
+func (s *localStoreTlsProvider) timerCallback() {
+	for {
+		select {
+		case <-s.stop:
+			break
+		case <-s.ticker.C:
+		}
+
+		var errorTime time.Time
+		if s.settings.ExpirationChecks.ErrorPeriod != 0 {
+			errorTime = time.Now().UTC().Add(s.settings.ExpirationChecks.ErrorPeriod)
+		} else {
+			errorTime = time.Now().UTC().AddDate(10, 0, 0)
+		}
+
+		period := s.settings.ExpirationChecks.WarningPeriod
+		if period != 0 {
+			expiring, expired, errs := s.Expiring(period)
+			s.reportErrors(errs)
+			if s.scope != nil {
+				s.scope.Gauge(metricCertsExpired).Update(float64(len(expired)))
+				s.scope.Gauge(metricCertsExpiring).Update(float64(len(expiring)))
+			}
+			s.logCerts(expired, true, errorTime)
+			s.logCerts(expiring, false, errorTime)
+		}
+	}
+}
+
+func (s *localStoreTlsProvider) logCerts(certs CertExpirationMap, expired bool, errorTime time.Time) {
+
+	for _, cert := range certs {
+		str := warningString(cert, expired)
```

Comment:
rename to createExpirationLogMsg or something. "warningString" implies its a string meant for the Warn log level